### PR TITLE
Refactor StorefrontContext

### PR DESCRIPTION
**Pull request by yuya-yuliya**
- refactor storefront part in StorefrontContext
- add ObjectProvider
- add ConnectEntityProvider
- rename Region models and move to Entities
- add RegionProfile and RegionMapper
- add RegionProfileTests
- split Commerce.Extension models to separate files
- rename models on FE

### CODE REVIEW

File: src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+// ReSharper disable CheckNamespace
+
+namespace Wooli.Foundation.Connect.Models
+{
+    using Glass.Mapper.Sc.Configuration.Attributes;
+
+    public partial class StorefrontModel
```
Comment (larionovadaria) [inline src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs:21]: `StorefrontSettings`

File: src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        private const string CountryRegionConfigurationQuery = "./*[@@templatename='" + Models.CountryRegionConfigurationModel.TemplateName + "']";
+        private const string CurrenciesDisplayAdjustmentsQuery = "./*[@@templatename='" + Models.CurrenciesDisplayAdjustmentsModel.TemplateName  + "']";
+        private const string PaymentConfigurationQuery = "./*[@@templatename='" + Models.PaymentConfigurationModel.TemplateName  + "']";
+
+        [SitecoreQuery(CountryRegionConfigurationQuery, IsRelative = true)]
+        public virtual CountryRegionConfigurationModel CountriesRegionsConfiguration { get; set; }
```
Comment (larionovadaria) [inline src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs:28]: `CountryRegionsSettings`

File: src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        // TODO: Use it in CurrencyProvider
+        [SitecoreQuery(CurrenciesDisplayAdjustmentsQuery, IsRelative = true)]
+        public virtual CurrenciesDisplayAdjustmentsModel CurrenciesDisplayAdjustments { get; set; }
```
Comment (larionovadaria) [inline src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs:32]: `CurrenciesDisplayAdjustmentsSettings`

File: src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        // TODO: Use this configuration in payment
+        [SitecoreQuery(PaymentConfigurationQuery, IsRelative = true)]
+        public virtual PaymentConfigurationModel PaymentConfiguration { get; set; }
```
Comment (larionovadaria) [inline src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs:36]: `PaymentSettings`

File: src/Foundation/Connect/code/Wooli.Foundation.Connect.csproj [hunk 5]
```diff
@@ -598,6 +597,11 @@
     <Compile Include="Mappers\Resolvers\MultilistFieldResolver.cs" />
     <Compile Include="Models\Catalog\StockStatus.cs" />
     <Compile Include="Models\Payment\PaymentClientTokenResult.cs" />
+    <Compile Include="Models\Storefront\CountryRegionConfigurationModel.cs" />
```
Comment (larionovadaria) [inline src/Foundation/Connect/code/Wooli.Foundation.Connect.csproj:600]: `Models\Settings\`

Comment (larionovadaria) [reply src/Foundation/Connect/code/Wooli.Foundation.Connect.csproj:600]: All the settings should have `Settings` postfix

File: src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs (hunk 1 above)
Comment (yuya-yuliya) [reply src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs:21]: This is extension of StorefrontModel class from Commerce.Generated

File: src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs (hunk 2 above)
Comment (yuya-yuliya) [reply src/Foundation/Connect/code/Models/Storefront/StorefrontModel.cs:28]: `CountryRegionConfigurationModel` is class from Commerce.Generated

File: src/Foundation/Connect/code/Wooli.Foundation.Connect.csproj (hunk 5 above)
Comment (yuya-yuliya) [reply src/Foundation/Connect/code/Wooli.Foundation.Connect.csproj:600]: This will be moved to `Models\Extensions\Storefront` in next commit

Comment (yuya-yuliya) [reply src/Foundation/Connect/code/Wooli.Foundation.Connect.csproj:600]: This models are from Commerce.Generated

**Pull request merged**